src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -92,113 +91,88 @@ private Address buildAddress(OfficeAddress dwpAddress) {
             return Optional.of(dwpMappings.getTestHmctsAddress());
         }
-        return findBenefitByShortName(benefitType)
-                .flatMap(b -> b.getOfficeMappings().apply(this, dwpIssuingOffice));
-    }
-
-    public Optional<OfficeMapping> carersAllowanceOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getCarersAllowance());
-    }
-
-    public Optional<OfficeMapping> dlaOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getDla());
-    }
-
-    public Optional<OfficeMapping> attendanceAllowanceOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getAttendanceAllowance());
-    }
-
-    public Optional<OfficeMapping> bereavementBenefitOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getBereavementBenefit());
-    }
-
-    public Optional<OfficeMapping> iidbOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getIidb());
-    }
-
-    public Optional<OfficeMapping> ucOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getUc());
-    }
-
-    public Optional<OfficeMapping> esaOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getEsa());
-    }
 
-    public Optional<OfficeMapping> pipOfficeMapping(String dwpIssuingOffice) {
-        String dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice,"(", ")"))
-                .orElse(dwpIssuingOffice.replaceAll("\\D+",""));
-
-        if (isEmpty(dwpIssuingOfficeStripped)) {
-            dwpIssuingOfficeStripped = dwpIssuingOffice;
+        //PIP search code
+        String dwpIssuingOfficeStripped = null;
+        if (findBenefitByShortName(benefitType).map(b -> b == PIP).orElse(false)) {
+            dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice, "(", ")"))
+                    .orElse(dwpIssuingOffice == null ? "" : dwpIssuingOffice.replaceAll("\\D+", ""));
         }
+        String dwpIssuingOfficeSearchKey = isEmpty(dwpIssuingOfficeStripped) ? dwpIssuingOffice : dwpIssuingOfficeStripped;
 
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOfficeStripped, dwpMappings.getPip());
-    }
+        List<OfficeMapping> mappings = findBenefitByShortName(benefitType)
+                .map(b -> b.getOfficeMappings().apply(this)).orElse(List.of());
 
-    public Optional<OfficeMapping> getDefaultDwpMappingByBenefitType(String benefitType) {
-        return Benefit.findBenefitByShortName(benefitType)
-                .flatMap(benefit -> benefit.getDefaultOfficeMapping().apply(this));
+        if (mappings.size() > 1) {
```

Comment:
I prefer returning early if there is one office, rather than doing the filtering for the dwpOffice here.
It's similar to a rule where you exit early on a method which doesn't fulful some criteria.

Reply:
Done